### DISCUSSION TITLE
How to Implement ReAct in Langchain in Langchain4j

### DISCUSSION THREAD

**Discussion by smileyboy2019** [langchain4j/langchain4j#1702] [Ideas]
How to Implement ReAct in Langchain in Langchain4j

**Comment by DDani** [comment]
That would be a nice example to have. 

By checking the original [ReAct prompt](https://smith.langchain.com/hub/hwchase17/react) on the documentation of Langchain I guess that it would be suitable to build an example by using an AI Service in some sort of a loop. 

However, I am not sure if it would be a good approach. Any thought?

**Comment by DDani** [comment]
After exploring what you mentioned about the AIService, I understand your point about the current models and their implementation of "embedded" ReAct, which ultimately refers to an iterative process until a valid response is found.

I don't have a clear use case, but in my business concept (a tool that provides recommendations to an employee based on their mood), the steps for an LLM would be:

Prompt: Given a person, provide recommendations to improve their well-being:

1. -> Thought: Identify who the employee is -> "Thought: I need to retrieve the employee's data."
2. <- Tool: Returns the basic ID of the employee.
3. -> Thought: With this info, tell me how the employee is doing.
4. <- Tool: Returns the employee's mood, with little detail.
5. -> Thought: I have limited information on the mood; I need to get the data from their latest survey.
6. <- Tool: Retrieves info from the latest survey.
7. -> Thought: I have all the information to make a recommendation...
8. ==> Emit result.

What I’m not clear on (I haven’t tested it, something like multiple calls) is whether a current AIService would call multiple tools (one after the other for each "thought") until it executes step 8.

**Reply by smileyboy2019** [reply]
![image](https://github.com/user-attachments/assets/778e4372-139d-4f9f-bcdd-22ec2b21f327)
https://github.com/dromara/liteflow
Similar workflow orchestration, but requires more intelligence

**Reply by smileyboy2019** [reply]
The main issue is how to execute in sequence and how to pass parameters
